aws-xray-recorder-sdk-sql/src/main/java/com/amazonaws/xray/sql/TracingStatement.java

```diff
@@ -0,0 +1,134 @@
+package com.amazonaws.xray.sql;
+
+import com.amazonaws.xray.AWSXRay;
+import com.amazonaws.xray.entities.Namespace;
+import com.amazonaws.xray.entities.Subsegment;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.sql.*;
+import java.util.HashMap;
+import java.util.Map;
+
+class TracingStatement {
+    static Statement decorate(Statement s, String invokedMethodName, Object[] invokedMethodArgs) {
+        return (Statement) Proxy.newProxyInstance(TracingConnection.class.getClassLoader(),
+                new Class[] { resolveProxiedInterface(invokedMethodName) },
+                new TracingStatementHandler(s, resolveSqlQuery(invokedMethodName, invokedMethodArgs)));
+    }
+
+    private static Class resolveProxiedInterface(String invokedMethodName) {
+        switch (invokedMethodName) {
+            case "prepareCall": return CallableStatement.class;
```

Comment:
convert hardcoded strings to variables please

Reply:
may extract them to a static constant class for reuse